### PR TITLE
More docs - starting to go through the extensions

### PR DESCRIPTION
**Pull request by adz**
Have a bit more time to get back into FSharpPlus so started in on some more API docs...

Now documents over Extensions pretty completely. Tried to follow conventions.

Please review the "Enumerator" file as it was a little tricky, choosing what to omit, etc.

### CODE REVIEW

**Comment by gusty** [comment]
I think it is because of the markdown.
I will have a look and try to identify which part breaks FSharp Formatting.
It happened to me before.

File: src/FSharpPlus/Extensions/Seq.fs [hunk 1]
```diff
@@ -20,6 +20,17 @@ module Seq =
...
 
+    /// <summary>Applies a sequence of functions to a sequence of values and concatenates them</summary>
+    /// <param name="f">The seq of functions.</param>
+    /// <param name="x">The seq of values.</param>
+    /// <returns>A seq concatenating the results from applying each function to each value</returns>
+    /// 
+    /// <example>
+    /// <code>
+    /// > Seq.apply [double; triple] [1; 2; 3];;  
+    /// val it : seq<int> = seq [2; 4; 6; 3; ...]
```
Comment (wallymathieu) [inline src/FSharpPlus/Extensions/Seq.fs:31]: You probably need to xml-escape the `<int>`

Comment (gusty) [reply src/FSharpPlus/Extensions/Seq.fs:31]: Yes, it was that. The generic brackets. At least in my case.

Comment (adz) [reply src/FSharpPlus/Extensions/Seq.fs:31]: Thanks!

File: src/FSharpPlus/Extensions/Array.fs [hunk 2]
```diff
@@ -6,6 +6,17 @@ module Array =
 
     open System
 
+    /// <summary>Applies an array of functions to an array of values and concatenates them</summary>
```
Comment (gusty) [inline src/FSharpPlus/Extensions/Array.fs:9]: Nitpicking: All sentences should end with a `.`

**Review by gusty — CHANGES_REQUESTED** [review]
Nice job, I just commented about conventions, if you think you can align them a bit.

File: src/FSharpPlus/Extensions/Map.fs [hunk 3]
```diff
@@ -9,12 +9,27 @@ module Map =
     open System.Linq
     #endif
 
+    /// NOTE: Missing from other dictionary like modules:
```
[the remark this answers is not in the document]
Comment (adz) [reply src/FSharpPlus/Extensions/Map.fs:12]: Yes indeed! Was a note to myself - I considered adding them right here, then decided to do it in separate PR.

File: src/FSharpPlus/Extensions/Dict.fs [hunk 4]
```diff
@@ -8,18 +8,49 @@ module Dict =
 
     #if !FABLE_COMPILER
     open System.Linq
+
+    /// Convert an IDictionary to an IReadOnlyDictionary
```
[the remark this answers is not in the document]
Comment (adz) [reply src/FSharpPlus/Extensions/Dict.fs]: Pretty sure it wasn't me, but I can see the inconsistency now. :+1:

File: src/FSharpPlus/Extensions/Array.fs (hunk 2 above)
Comment (adz) [reply src/FSharpPlus/Extensions/Array.fs]: Have done, and also reviewed all doc strings in Extensions.

Note: I generally treat doc summaries as a title without a full stop, however even I was inconsistent as I copied some existing comments sometimes!

File: src/FSharpPlus/Extensions/Seq.fs [hunk 5]
```diff
@@ -37,17 +36,24 @@ module Seq =
...
-    /// keys tupled with values. Each key contains an array of all adjacent elements that match 
-    /// to this key, therefore keys are not unique but they can't be adjacent
-    /// as each time the key changes, a new group is yield.</summary>
+    /// <summary>
+    /// Chunks the seq up into groups with the same projected key by applying
+    /// the key-generating projection function to each element and yielding a sequence of 
+    /// keys tupled with values.
+    /// </summary>
+    ///
+    /// <remarsk>
```
Comment (gusty) [inline src/FSharpPlus/Extensions/Seq.fs:45]: This typo is making the build to fail.

**Review by gusty — CHANGES_REQUESTED** [review]
Looks good. Existing docs are not always standard, but we can align them as we go.

**Review by gusty — APPROVED** [review]
Excellent work !

**Pull request merged**